streamwebs_frontend/streamwebs/models.py

```diff
@@ -233,6 +232,73 @@ class Meta:
         verbose_name_plural = 'Water Quality'
 
 
+class CameraPointManager(models.Manager):
+
+    def create_camera_point(self, site, cp_date, created_by='', latitude=None,
+                            longitude=None, map_datum='', description=''):
+
+        return self.create(site=site, cp_date=cp_date, created_by=created_by,
+                           latitude=latitude, longitude=longitude,
+                           map_datum=map_datum, description=description)
+
+
+class CameraPoint(models.Model):
+    site = models.ForeignKey(Site, on_delete=models.CASCADE, null=True)
+    cp_date = models.DateField(default=datetime.date.today)
+    created_by = models.CharField(max_length=255, blank=True)
+    latitude = models.DecimalField(default=0, max_digits=9, decimal_places=6,
+                                   blank=True, null=True)
+    longitude = models.DecimalField(default=0, max_digits=9, decimal_places=6,
+                                    blank=True, null=True)
+    map_datum = models.CharField(max_length=255, blank=True)
+    description = models.TextField(blank=True)
+
+    camera_points = CameraPointManager()
+
+    def __str__(self):
+        return ('Camera point ' + str(self.id) + ' for site ' +
+                self.site.site_name)
+
+    class Meta:
+        verbose_name = 'Camera Point'
+        verbose_name_plural = 'Camera Points'
+
+
+class PhotoPointManager(models.Manager):
+
+    def create_photo_point(self, camera_point, pp_date, compass_bearing,
+                           distance, camera_height, photo_filename='',
+                           photo=None, notes=''):
+
+        return self.create(camera_point=camera_point, pp_date=pp_date,
+                           compass_bearing=compass_bearing, distance=distance,
+                           camera_height=camera_height,
+                           photo_filename=photo_filename, photo=photo,
+                           notes=notes)
+
+
+class PhotoPoint(models.Model):
+    camera_point = models.ForeignKey(CameraPoint, on_delete=models.CASCADE,
+                                     null=True, related_name='camera_point')
+    pp_date = models.DateField(default=datetime.date.today)
+    compass_bearing = models.PositiveSmallIntegerField()
+    distance = models.DecimalField(max_digits=3, decimal_places=0)
+    camera_height = models.DecimalField(max_digits=3, decimal_places=0)
+    photo_filename = models.CharField(max_length=255, blank=True)
```

Comment:
Very true. I included that field because the original printable data sheet
does have a line for that. I just put in there because I thought maybe
someone who brought their tablet into the field to record data with, but
was taking pictures with another camera might want to note the filename
first so that they can upload the correct picture later. But they have
labeled boards in their photos, so I can delete this field.

On Wed, Jul 13, 2016 at 4:42 PM, Kennric <notifications@github.com> wrote:

> In streamwebs_frontend/streamwebs/models.py
> <https://github.com/osuosl/streamwebs/pull/60#discussion_r70727778>:
>
> > +
> > +        return self.create(camera_point=camera_point, pp_date=pp_date,
> > +                           compass_bearing=compass_bearing, distance=distance,
> > +                           camera_height=camera_height,
> > +                           photo_filename=photo_filename, photo=photo,
> > +                           notes=notes)
> > +
> > +
> > +class PhotoPoint(models.Model):
> > +    camera_point = models.ForeignKey(CameraPoint, on_delete=models.CASCADE,
> > +                                     null=True, related_name='camera_point')
> > +    pp_date = models.DateField(default=datetime.date.today)
> > +    compass_bearing = models.PositiveSmallIntegerField()
> > +    distance = models.DecimalField(max_digits=3, decimal_places=0)
> > +    camera_height = models.DecimalField(max_digits=3, decimal_places=0)
> > +    photo_filename = models.CharField(max_length=255, blank=True)
>
> Do you need the filename when you have ImageField? You should be able to
> get the filename from the imagefield
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/osuosl/streamwebs/pull/60/files/2c7b3b9fd948e8d3cda5715503b960379cbc2fb6#r70727778>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIu_SSP3AELoB3oBxOcEI7ezlYGqcIV0ks5qVXfIgaJpZM4JGh7J>
> .
>